tests/e2e/e2e_pod_client_server_test.go

```diff
@@ -25,26 +26,33 @@ var _ = OSMDescribe("Test HTTP traffic from 1 pod client -> 1 pod server",
 	func() {
 		Context("Test traffic flowing from client to server with a Kubernetes Service for the Source: HTTP", func() {
 			withSourceKubernetesService := true
-			testTraffic(withSourceKubernetesService)
+			testTraffic(withSourceKubernetesService, PodCommandDefault)
 		})
 
 		Context("Test traffic flowing from client to server without a Kubernetes Service for the Source: HTTP", func() {
 			// Prior iterations of OSM required that a source pod belong to a Kubernetes service
 			// for the Envoy proxy to be configured for outbound traffic to some remote server.
 			// This test ensures we test this scenario: client Pod is not associated w/ a service.
 			withSourceKubernetesService := false
-			testTraffic(withSourceKubernetesService)
+			testTraffic(withSourceKubernetesService, PodCommandDefault)
+		})
+
+		Context("Test traffic flowing from client to a server with a podIP bind", func() {
```

Comment:
Let's just modify one of the existing tests to bind to the pod IP vs adding another test here. This gives us the coverage we need without increasing the CI time. This test is also not the best test to change, as it is already testing an additional scenario of client with/without k8s service. Could you update a different test, say `e2e_deployment_client_server_test` to simply bind to the pod IP.

Reply:
Binding to the pod IP is an explicit behavior that I would think we'd want to test explicitly in a separate test rather than silently attaching it to another test. I picked this test because it's testing networking scenarios: client-server with service and client-server without service. Does it not make sense to test other networking behavior here in the simplest case (i.e. 2 pods)?

Reply:
Initially, I was going to make this a completely separate test, but when I saw there was already a `testTraffic` function generic enough for my use-case, I decided to use it. Regardless of `e2e_pod_client_server_test.go`'s original intent, the Describe block's title is: "Test HTTP traffic from 1 pod client -> 1 pod server". That's pretty generic, no? There's no other test that's this simple (from what I can tell), so I think it makes sense to test pod-to-pod connectivity in this way.

Also, just because the code change is related to iptables, doesn't necessarily mean that's where the e2e test should be located. The whole point of an integration/e2e test is to think from a customer's perspective. When I think of how a bug in this area would present itself, pod-to-pod connectivity is what I come up with (as you and I have both seen from the IcM)

Reply:
Just to share more context regarding this test, it started out as a test to verify HTTP traffic between pods. Then we extended this to test the scenario where a client can have a service associated with it, just so we don't add more tests that have an extremely narrow scope. At the very least if you want to extend this test to further connectivity scenarios, please rename it to something such as `e2e_client_server_connectivity_test.go`. Previously, this test was differentiating itself from `e2e_deployment_client_server_test` by creating Pods rather than Deployment objects, so the meaning of this test is being changed here, which is why I am suggesting making it explicit. Otherwise, it seems like adding one more scenario to an unrelated test (which is something I suggested to reduce the CI time).

Reply:
Ah, given your context on the evolution of the test file, I now understand your position. I think renaming the file makes sense given its semantics. 👍🏾